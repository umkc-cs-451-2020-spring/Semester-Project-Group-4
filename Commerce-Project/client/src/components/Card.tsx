import React from 'react';
import { Card, CardContent, Typography, IconButton, Tooltip, Paper } from '@material-ui/core';
import { makeStyles, createStyles } from '@material-ui/styles';
import Close from '@material-ui/icons/Close';
import { Theme } from './Theme';

const useStyles = makeStyles(() =>
  createStyles({
    card: {
      width: 250,
      height: '7rem',
    },
    contentArea: {
      paddingTop: '.5rem',
      paddingRight: '.5rem',
      paddingLeft: '1rem'
    },
    iconColor: {
      color: Theme.palette.error.main,
      padding: 'unset',
      marginLeft: '67%'
    },
    title: {
      color: Theme.palette.error.main
    },
    closeButton: {
      flexDirection: 'row-reverse',
      padding: 'unset'
    },
    description: {
      paddingLeft: 'unset'
    },
    row: {
      display: 'inline-block',
      paddingRight: '1rem',
      paddingBottom: '1rem'
    }
  })
);

export interface NotificationProps {
  message: string;
  onClick: any;
}

export const NotificationCard = (props: NotificationProps) => {
  const { card, iconColor, contentArea, description, title, row } = useStyles();
  const { message, onClick } = props;

  return (
    <Paper elevation={0} className={row}>
      <Card className={card} variant="outlined">
        <CardContent className={contentArea}>
          <Typography variant="h5" component="h2" className={title}>
            Alert
          <Tooltip title='Remove'>
              <IconButton className={iconColor} onClick={onClick}>
                <Close />
              </IconButton>
            </Tooltip>
          </Typography>
          <CardContent className={description}>
            <Typography variant="body1" component="p">
              {message}
            </Typography>
          </CardContent>
        </CardContent>
      </Card>
    </Paper>
  )
}