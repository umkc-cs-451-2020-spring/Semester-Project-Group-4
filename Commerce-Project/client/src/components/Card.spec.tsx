import React from 'react';
import { shallow } from 'enzyme';
import { NotificationCard } from './Card';

describe('Notification Card', () => {

  it('Render the Notification Card', () => {
    const component = shallow(<NotificationCard message='test' />);

    expect(component).toHaveLength(1);
    expect(component).toMatchSnapshot();
  });
});