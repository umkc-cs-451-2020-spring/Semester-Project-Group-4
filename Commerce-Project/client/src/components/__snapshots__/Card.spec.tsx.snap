// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification Card Render the Notification Card 1`] = `
<WithStyles(ForwardRef(Paper))
  className="makeStyles-row-7"
  elevation={0}
>
  <WithStyles(ForwardRef(Card))
    className="makeStyles-card-1"
    variant="outlined"
  >
    <WithStyles(ForwardRef(CardContent))
      className="makeStyles-contentArea-2"
    >
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-title-4"
        component="h2"
        variant="h5"
      >
        Alert
        <WithStyles(ForwardRef(Tooltip))
          title="Remove"
        >
          <WithStyles(ForwardRef(IconButton))
            className="makeStyles-iconColor-3"
          >
            <CloseIcon />
          </WithStyles(ForwardRef(IconButton))>
        </WithStyles(ForwardRef(Tooltip))>
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(CardContent))
        className="makeStyles-description-6"
      >
        <WithStyles(ForwardRef(Typography))
          component="p"
          variant="body1"
        >
          test
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(CardContent))>
    </WithStyles(ForwardRef(CardContent))>
  </WithStyles(ForwardRef(Card))>
</WithStyles(ForwardRef(Paper))>
`;
